File: challenges/3-tipos-de-triangulo/java/Neg0023/TiposDeTriangulo.java
public class TiposDeTriangulo {
    public static void main(String[] args) {
        double a = Double.parseDouble(args[0]);
        double b = Double.parseDouble(args[1]);
        double c = Double.parseDouble(args[2]);

        System.out.printf("Entrada: %.1f %.1f %.1f%n", a, b, c);

        double troca;

        if (b > a && b >= c) {
            troca = a;
            a = b;
            b = troca;
        } else if (c > a && c >= b) {
            troca = a;
            a = c;
            c = troca;
        }

        if (c > b) {
            troca = b;
            b = c;
            c = troca;
        }


        if (a >= (b + c)) {
            System.out.println("NAO FORMA TRIANGULO");
        } else if ((a * a) == (b * b) + (c * c)) {
            System.out.println("TRIANGULO RETANGULO");
        } else {
            boolean b1 = (a == b && b != c) || (a != b && b == c);
            if ((a * a) > (b * b) + (c * c)) {
                System.out.println("TRIANGULO OBTUSANGULO");
                if (a == b && b == c) {
                    System.out.println("TRIANGULO EQUILATERO");
                } else if (b1) {
                    System.out.println("TRIANGULO ISOSCELES");
                }
            } else if ((a * a) < (b * b) + (c * c)) {
                System.out.println("TRIANGULO ACUTANGULO");
                if (a == b && b == c) {
                    System.out.println("TRIANGULO EQUILATERO");
                }else if (b1) {
                    System.out.println("TRIANGULO ISOSCELES");
                }
            }
        }
    }
}

File: challenges/3-tipos-de-triangulo/typescript/clerijr/tiposDeTriangulo.ts
let arg_A: number = Number(process.argv.slice(2, 3))
let arg_B: number = Number(process.argv.slice(3, 4))
let arg_C: number = Number(process.argv.slice(4, 5))

let response: string = ''

if (arg_A >= arg_B + arg_C) response += 'NAO FORMA TRIANGULO\n'
if (arg_A ** 2 == arg_B ** 2 + arg_C ** 2) response += 'TRIANGULO RETANGULO\n'
if (arg_A ** 2 > arg_B ** 2 + arg_C ** 2) response += 'TRIANGULO OBTUSANGULO\n'
if (arg_A ** 2 < arg_B ** 2 + arg_C ** 2) response += 'TRIANGULO ACUTANGULO\n'
if (arg_A == arg_B && arg_B == arg_C) {
    response += 'TRIANGULO EQUILATERO\n'
} else if (arg_A == arg_B && arg_B == arg_C && arg_A == arg_C) { 
    response += 'TRIANGULO ISOSCELES\n' 
}

console.log(response)